fenicsprecice/fenicsprecice.py

```diff
@@ -273,9 +286,47 @@ def initialize(self, coupling_subdomain, mesh, function_space, write_function=No
             Recommended time step value from preCICE.
         """
 
-        coords = function_space.tabulate_dof_coordinates()
+        write_function_space, write_function = None, None
+        if type(write_object) is Function:
+            write_function_space = write_object.function_space()
+            write_function = write_object
+        elif type(write_object) is FunctionSpace:
+            write_function_space = write_object
+            write_function = None
+        elif write_object is None:
+            print("Participant {} is read-only participant".format(self._config.get_participant_name()))
+        else:
+            raise Exception("Given write object is neither of type dolfin.functions.function.Function or "
+                            "dolfin.functions.functionspace.FunctionSpace")
```

Comment:
```suggestion
            raise Exception("Given write_object is neither of type dolfin.functions.function.Function or "
                            "dolfin.functions.functionspace.FunctionSpace. Please provide a Function, if you "
                            "want to initialize coupling data (via precice.initialize_data()) according to this "
                            "function. A FunctionSpace is sufficient, if you don't want to initialize data. If you 
                            "want to do one-way coupling where this participant is only reading data, just "
                            "leave the write_object blank.")
```
Might be a bit verbose, but then the user really knows what's going on.